### PR TITLE
Remove CosmosDB database provider

### PR DESCRIPTION
**Pull request by rynowak**
# Description

This is part of a refactor of the data layer to simplify the design. The CosmosDB implementation is complicated and unused so we're removing it.

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).

## Contributor checklist
Please verify that the PR meets the following requirements, where applicable:

- [ ] An overview of proposed schema changes is included in a linked GitHub issue.
- [ ] A design document PR is created in the [design-notes repository](https://github.com/radius-project/design-notes/), if new APIs are being introduced.
- [ ] If applicable, design document has been reviewed and approved by Radius maintainers/approvers.
- [ ] A PR for the [samples repository](https://github.com/radius-project/samples) is created, if existing samples are affected by the changes in this PR.
- [ ] A PR for the [documentation repository](https://github.com/radius-project/docs) is created, if the changes in this PR affect the documentation or any user facing updates are made.
- [ ] A PR for the [recipes repository](https://github.com/radius-project/recipes) is created, if existing recipes are affected by the changes in this PR.

### CODE REVIEW

File: cmd/applications-rp/radius-cloud.yaml [hunk 1]
```diff
@@ -1,56 +0,0 @@
...
-  instance: "https://login.windows.net"
-  tenantId: "common"
-  armEndpoint: "https://management.azure.com:443"
-  audience: "https://management.core.windows.net"
-  pemCertPath: "/var/certs/rp-aad-app.pem"
-storageProvider:
-  provider: "cosmosdb"
-  cosmosdb:
-    # Create your own SQL API Cosmos DB account and set url in this configuration or to RADIUS_STORAGEPROVIDER_COSMOSDB_URL environment variable
-    url: https://radius-eastus-test.documents.azure.com:443/
```
Comment (rynowak) [inline cmd/applications-rp/radius-cloud.yaml:16]: I thought it best to just remove this file. There's a bunch of Azure-specific stuff in here that's not relevant. I can't think of a scenario where this is used.

**Review by ytimocin — APPROVED** [review]

**Review by lakshmimsft — APPROVED** [review]

File: pkg/corerp/README.md [hunk 2]
```diff
@@ -43,14 +42,12 @@
...
     ```
 
-1. To get information about an environment, here is an example curl command:
+4. To get information about an environment, here is an example curl command:
 
     ```
     curl --location --request GET 'http://localhost:8080/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/radius-test-rg/providers/Applications.Core/environments/?api-version=2023-10-01-preview'
     ```
 
-1. You should also be able to see all changes in Cosmos DB
```
Comment (kachawla) [inline pkg/corerp/README.md:52]: Should we update this to point to etcd (and later to postgres once the work completes) instead of deleting?

Comment (rynowak) [reply pkg/corerp/README.md:52]: There's no straightforward way to do this with etcd. 

What do you think about making this part of the postgres work? @superbeeny and I talked about updating the dev instructions as part of that.

Comment (kachawla) [reply pkg/corerp/README.md:52]: that sounds good

**Pull request merged**